### PR TITLE
Add a new machineset for infra components

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-1784

Tested locally and it works:

```
oc --context build01 get node ip-10-0-134-111.ec2.internal
NAME                           STATUS   ROLES          AGE   VERSION
ip-10-0-134-111.ec2.internal   Ready    infra,worker   50s   v1.19.0+7070803

oc --context build01 get node ip-10-0-134-111.ec2.internal -o yaml | yq -y '.spec.taints'
- effect: NoSchedule
  key: node-role.kubernetes.io/infra

c --context build01 get node ip-10-0-134-111.ec2.internal -o yaml | yq -y '.metadata.labels' | grep infra
node-role.kubernetes.io: infra
node-role.kubernetes.io/infra: ''

```

/cc @alvaroaleman

### CODE REVIEW

**Comment by hongkailiu** [comment]
/hold

more changes to come, like autoscaler, noschedule

**Comment by alvaroaleman** [comment]
just testing ci...
/test boskos-config

File: clusters/build-clusters/01_cluster/openshift-ingress-operator/default_ingresscontroller.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#We put only the customized fields in git
```
Comment (alvaroaleman) [inline clusters/build-clusters/01_cluster/openshift-ingress-operator/default_ingresscontroller.yaml:1]: Can we put this into the "common without api ci folder" as this is not specfic to any one cluster?

Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/openshift-ingress-operator/default_ingresscontroller.yaml:1]: Add more fields we touched.
It is specific to build01 (my bad naming strategy slows us down).

`build02` has not yet set up infra? Do we want to do it all together?
More over, do we want to do this to vsphere as well?

Comment (alvaroaleman) [reply clusters/build-clusters/01_cluster/openshift-ingress-operator/default_ingresscontroller.yaml:1]: We want this for build02, yes. Vsphere is a weird one as we do not own it

File: clusters/build-clusters/01_cluster/openshift-image-registry/cluster_config.yaml [hunk 2]
```diff
@@ -4,7 +4,13 @@ kind: Config
 metadata:
   name: cluster
 spec:
+  nodeSelector:
+    node-role.kubernetes.io/infra: ""
```
Comment (alvaroaleman) [inline clusters/build-clusters/01_cluster/openshift-image-registry/cluster_config.yaml:8]: Nit: Can we use something that does not have `kubernetes.io` in the name, as this label is not official at all and specific to what we do `dptp.openshift.io` or sth)?

File: clusters/build-clusters/01_cluster/openshift-machine-api/admin_build01-9hdwj-worker-us-east-1a-infra_machine_set.yaml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      taints:
+      - effect: NoSchedule
+        key: node-role.kubernetes.io/infra
+      providerSpec:
+        value:
+          ami:
+            id: ami-06f85a7940faa3217
+          apiVersion: awsproviderconfig.openshift.io/v1beta1
+          blockDevices:
+          - ebs:
```
Comment (alvaroaleman) [inline clusters/build-clusters/01_cluster/openshift-machine-api/admin_build01-9hdwj-worker-us-east-1a-infra_machine_set.yaml:37]: Is this needed? I would expect the infra node to have minimal disk io

File: clusters/build-clusters/01_cluster/openshift-ingress-operator/default_ingresscontroller.yaml (hunk 1 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/openshift-ingress-operator/default_ingresscontroller.yaml:1]: If that is not for `vsphere`, then we cannot put it into https://github.com/openshift/release/tree/master/clusters/build-clusters/common_except_app.ci

The one for build02 is another PR.

File: clusters/build-clusters/01_cluster/openshift-image-registry/cluster_config.yaml (hunk 2 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/openshift-image-registry/cluster_config.yaml:8]: I guess that the name make it shows up when we do

```
oc --context build01 get node ip-10-0-134-111.ec2.internal
NAME                           STATUS   ROLES          AGE    VERSION
ip-10-0-134-111.ec2.internal   Ready    infra,worker   155m   v1.19.0+7070803
```

I feel it is a nice-to-have thing.

File: clusters/build-clusters/01_cluster/openshift-machine-api/admin_build01-9hdwj-worker-us-east-1a-infra_machine_set.yaml (hunk 3 above)
Comment (hongkailiu) [reply clusters/build-clusters/01_cluster/openshift-machine-api/admin_build01-9hdwj-worker-us-east-1a-infra_machine_set.yaml:37]: Thought about that.
Let me modify.

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**